### PR TITLE
Fixes #12099 - Template locked? relying on rake makes tests fail

### PR DESCRIPTION
**Pull request by dLobatog**
app/models/template.rb has a method locked? that relies on
!Foreman.in_rake? to bypass the locked validation when running some
operations through Rake such as migrations or cloning templates. On
Rails 4 rake test is also interpreted as being 'in rake' and tests fail
when running them through rake test but not when running template tests
individually.

### CODE REVIEW

**Comment by shlomizadok** [comment]
[test]

**Comment by unorthodoxgeek** [comment]
:+1: good for merge IMHO

File: test/unit/provisioning_template_test.rb [hunk 1]
```diff
@@ -85,10 +85,26 @@ class ProvisioningTemplateTest < ActiveSupport::TestCase
     assert FactoryGirl.create(:provisioning_template, :locked => true)
   end
 
-  test "should not edit a locked template" do
-    tmplt = templates(:locked)
-    tmplt.name = "something else"
-    refute_valid tmplt, :base, /is locked/
+  context 'locked templates outside of rake' do
+    setup do
+      Foreman.stubs(:in_rake?).returns(false)
```
Comment (domcleal) [inline test/unit/provisioning_template_test.rb]: Nitpick, prefer `.expects` just to ensure we're exercising the code as we expect.  You'd also need to add `.at_least_once` to allow it to call in_rake? twice.

**Comment by dLobatog** [comment]
Fair enough, updated to expect the `in_rake` call :smile: !

**Comment by domcleal** [comment]
Merged as 1cff242, thanks @dLobatog.

**Pull request closed without merging**